packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1330,4 +1333,253 @@ void main() {
+      bool wrap = false;
+
+      return MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (BuildContext context, void Function(void Function()) setState) {
+                Future<void> changeWrap() async {
+                  await Future<void>.delayed(const Duration(milliseconds: 50));
+
+                  setState(() {
+                    wrap = !wrap;
+                  });
+                }
+
+                Widget child = InkWell(
+                  onTapDown: (_) {
+                    if (onTapDownChangeWrap)
+                      changeWrap();
+                  },
+                  onTap: () {
+                    if (onTapChangeWrap)
+                      changeWrap();
+                  },
+                  child: Container(
+                    height: 160,
+                    alignment: Alignment.center,
+                  ),
+                );
+
+                child = Container(
+                  key: key,
+                  child: child,
+                );
+
+                if (wrap) {
+                  child = Container(
+                    margin: const EdgeInsets.only(top: 320),
+                    child: child,
+                  );
+                }
+
+                return child;
+              },
+            ),
+          ),
+        ),
+      );
+    });
+  });
+
+  testWidgets('When InkWell/Ancestor has a GlobalKey and ancestor Material is replaced, splash should stop.', (WidgetTester tester)async {
+    final Key key = GlobalKey();
+
+    bool replaced = false;
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: StatefulBuilder(builder: (BuildContext context, void Function(void Function()) setState) {
+          Future<void> changeReplace() async {
+            await Future<void>.delayed(const Duration(milliseconds: 50));
+
+            setState(() {
+              replaced = !replaced;
+            });
+          }
+
+          return Material(
+            key: ValueKey<bool>(replaced),
+            child: InkWell(
+              key: key,
+              onTap: () {
+                changeReplace();
+              },
+            ),
+          );
+        }),
+      ),
+    ));
+
+    await tester.tap(find.byType(InkWell));
+    await tester.pump(const Duration(milliseconds: 60));
+
+    final PaintPattern paintPattern = paints..circle();
+    expect(
+      Material.of(tester.element(find.byType(InkWell)))! as RenderBox,
+      isNot(paintPattern),
+    );
+  });
+
+  testWidgets('When InkWell/Ancestor has a GlobalKey and ancestor Material is replaced, highlight should always be maintained.', (WidgetTester tester) async {
+    final Key key = GlobalKey();
+    const Color hoverColor = Color(0xff00ff00);
+
+    int onHoverCount = 0;
+    int callChangeReplaceCount = 0;
+    bool replaced = false;
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
```

Comment:
tests should be as simple as possible. Do you need the Scaffold here for example?